### PR TITLE
img: fix srcset pattern and src default value

### PR DESCRIPTION
**Pull request by Kdecherf**
The first change updates srcset pattern to support URL without any descriptor as it is supported by the standard, see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Img#attr-srcset

The second change sets the default img src value to an empty gif. Without this change, HTMLawed adds `src="src"` to invalid tags which can lead to unexpected behaviors on browsers or http client (_e.g. Graby_).

Note: this PR is based on #8

### CODE REVIEW

File: htmLawed.php [hunk 1]
```diff
@@ -814,7 +814,7 @@ function hl_tag($t)
                 $v = str_replace('­', ' ', (false !== strpos($v, '&') ? str_replace(['&#xad;', '&#173;', '&shy;'], ' ', $v) : $v)); // double-quoted char: soft-hyphen; appears here as "­" or hyphen or something else depending on viewing software
                 if ('srcset' === $k) {
                     $v2 = '';
-                    $pattern = "/(?:[^\"'\s]+\s*(?:\d+w|\d+(?:\.\d+)?x)+)/";
+                    $pattern = "/(?:[^\"',\s]+(?:\s(?:\d+w|\d+(?:\.\d+)?x))?)/";
```
Comment (jtojnar) [inline htmLawed.php]: Ugh, this really needs test coverage. If I am reading https://html.spec.whatwg.org/multipage/images.html#srcset-attribute right, it should be `\s+`.

File: htmLawed.php [hunk 2]
```diff
@@ -874,7 +874,7 @@ function hl_tag($t)
     }
 
     // rqd attr
-    static $eAR = ['area' => ['alt' => 'area'], 'bdo' => ['dir' => 'ltr'], 'command' => ['label' => ''], 'form' => ['action' => ''], 'img' => ['src' => '', 'alt' => 'image'], 'map' => ['name' => ''], 'optgroup' => ['label' => ''], 'param' => ['name' => ''], 'style' => ['scoped' => ''], 'textarea' => ['rows' => '10', 'cols' => '50']];
+    static $eAR = ['area' => ['alt' => 'area'], 'bdo' => ['dir' => 'ltr'], 'command' => ['label' => ''], 'form' => ['action' => ''], 'img' => ['src' => 'data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw==', 'alt' => 'image'], 'map' => ['name' => ''], 'optgroup' => ['label' => ''], 'param' => ['name' => ''], 'style' => ['scoped' => ''], 'textarea' => ['rows' => '10', 'cols' => '50']];
```
Comment (jtojnar) [inline htmLawed.php:877]: Sounds good to me, reading https://html.spec.whatwg.org/multipage/embedded-content.html#attr-img-src

File: htmLawed.php (hunk 1 above)
Comment (Kdecherf) [reply htmLawed.php]: Hm indeed, I'll rework it and I'll try to add tests

Comment (Kdecherf) [reply htmLawed.php]: Fixed the regex in https://github.com/fossar/HTMLawed/pull/9/commits/51698deb86146d82816d13b7812b2cfad69c9f6e

Added tests for this case along Github Actions workflow in https://github.com/fossar/HTMLawed/pull/9/commits/e012d637ce982a481b8f5c714800e304a12b0c1e

Comment (Kdecherf) [reply htmLawed.php]: Poke @j0k3r

**Comment by Kdecherf** [comment]
@j0k3r @jtojnar tests on php 7.2 and php 7.3 are failing because of a dependency issue.

Knowing that php < 7.4 are not supported anymore, could be bump the minimum required php version to 7.4?

File: .github/workflows/continuous-integration.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    steps:
+      - name: "Checkout"
+        uses: "actions/checkout@v2"
+
+      - name: "Install PHP"
+        uses: "shivammathur/setup-php@v2"
+        with:
+          php-version: "${{ matrix.php }}"
+          coverage: none
+          tools: pecl, composer:v1
```
Comment (j0k3r) [inline .github/workflows/continuous-integration.yml]:
```suggestion
          tools: pecl, composer:v2
```

File: .github/workflows/continuous-integration.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+          extensions: tidy
+          ini-values: "date.timezone=Europe/Paris"
+
+      - name: "Install dependencies with Composer"
+        uses: "ramsey/composer-install@v1"
```
Comment (j0k3r) [inline .github/workflows/continuous-integration.yml]:
```suggestion
        uses: "ramsey/composer-install@v2"
```

**Review by j0k3r — COMMENTED** [review]
You should stay on `v3.4.0` for `friendsofphp/php-cs-fixer` to avoid the 7.4 bump.
As it's a dev deps, I think it's preferable to keep 7.2 compats.
Maybe you should also lower PHPUnit?

**Comment by Kdecherf** [comment]
@j0k3r I've updated the lock file to keep support of PHP 7.2 and 7.3.

Interesting thing to note: I was forced to call `composer update` with a PHP 7.2 install to have compatible dependencies.

**Comment by j0k3r** [comment]
@Kdecherf https://getcomposer.org/doc/06-config.md#platform 😉

File: .github/workflows/continuous-integration.yml (hunk 3 above)
Comment (j0k3r) [reply .github/workflows/continuous-integration.yml]: Would like to keep `v1` or you forget it?

Comment (Kdecherf) [reply .github/workflows/continuous-integration.yml]: Ahh good catch, just forgot it. Fixed.

**Comment by Kdecherf** [comment]
> @Kdecherf https://getcomposer.org/doc/06-config.md#platform wink

@j0k3r not sure of the usage of the property, should I fix it before merging this PR or is it fine for now?

**Comment by j0k3r** [comment]
I still prefer to define it so nobody will have error if they update the proches using a higher version of PHP.
Put the version 7.2.34

**Comment by j0k3r** [comment]
Looks good to me!

**Review by j0k3r — APPROVED** [review]

**Pull request merged**